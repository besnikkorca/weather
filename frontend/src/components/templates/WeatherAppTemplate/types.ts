export interface City {
  id: number;
  name: string;
}
export interface Props {
  cities: City[];
}
