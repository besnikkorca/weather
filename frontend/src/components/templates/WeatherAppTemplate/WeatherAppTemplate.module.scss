// This selector is more specific to override
// the padding applied from the .container class
div.contentWrapper {
  padding: 4rem 0.5rem;

  @media only screen and (min-width: 1224px) {
    & {
      padding: 2rem 0.5rem;
    }
  }
  /* Extra Large screens ----------- */
  @media only screen and (min-width: 1921px) {
    & {
      padding: 4rem 0.5rem;
    }
  }
}
