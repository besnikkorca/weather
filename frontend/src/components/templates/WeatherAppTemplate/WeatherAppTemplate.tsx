import { Component } from 'react';
import WeatherCard from 'molecules/WeatherCard';
import WeatherList from 'molecules/WeatherList';
import WeatherForecast from 'organisms/WeatherForecast';
import Container from 'atoms/Container';
import Header from 'organisms/Header';
import { Props } from './types';

export default class WeatherAppTemplate extends Component<Props> {
  state = {
    active: 'Ottawa',
  };

  setActive = (city: string) => {
    this.setState({ active: city });
  };

  getCityId = () => {
    return this.props.cities.find(({ name }) => name === this.state.active)?.id;
  };

  render() {
    const cityId = this.getCityId();
    return (
      <Container full>
        <Container maxWidth>
          <Header active={this.state.active} setActive={this.setActive} />
          {cityId ? (
            <WeatherForecast cityId={cityId} />
          ) : (
            <Container>Please select a city!</Container>
          )}
        </Container>
      </Container>
    );
  }
}
