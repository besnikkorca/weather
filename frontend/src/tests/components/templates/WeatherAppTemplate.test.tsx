import { render } from '@testing-library/react';
import WeatherAppTemplate from 'templates/WeatherAppTemplate';
import mockUseQuery from '../services/query/mockUseQuery';

const cities = [
  {
    id: 1,
    name: 'Ottawa',
  },
  {
    id: 2,
    name: 'Moscow',
  },
];

describe('WeatherAppTemplate', () => {
  beforeEach(() => {
    mockUseQuery((key, fn, options) => ({
      data: [],
      isLoading: false,
      isError: false,
    }));
  });

  it('should render', () => {
    const { container } = render(<WeatherAppTemplate cities={cities} />);

    expect(container).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const view = render(<WeatherAppTemplate cities={cities} />);

    expect(view).toMatchSnapshot();
  });

  it('renders correctly with no cities', () => {
    const view = render(<WeatherAppTemplate cities={[]} />);

    expect(view).toMatchSnapshot();
  });

  it('renders correctly with one city', () => {
    const view = render(<WeatherAppTemplate cities={[cities[0]]} />);

    expect(view).toMatchSnapshot();
  });

  afterEach(() => {
    jest.resetAllMocks();
  });
});
