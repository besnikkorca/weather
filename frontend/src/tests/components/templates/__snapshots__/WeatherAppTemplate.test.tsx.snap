// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherAppTemplate renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container full transparentBG contentWrapper"
      >
        <div
          class="container transparentBG maxWidthLarge"
        >
          <header
            class="header"
          >
            <nav>
              <ul
                class="list"
              >
                <li
                  class="listItem active"
                >
                  Ottawa
                </li>
                <li
                  class="listItem notActive"
                >
                  Moscow
                </li>
                <li
                  class="listItem notActive"
                >
                  Tokyo
                </li>
              </ul>
            </nav>
          </header>
          <div
            class="container flexColumn transparentBG border borderRadius shadow container"
          >
            <p
              class="default large weightNormal"
            >
              Error
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container full transparentBG contentWrapper"
    >
      <div
        class="container transparentBG maxWidthLarge"
      >
        <header
          class="header"
        >
          <nav>
            <ul
              class="list"
            >
              <li
                class="listItem active"
              >
                Ottawa
              </li>
              <li
                class="listItem notActive"
              >
                Moscow
              </li>
              <li
                class="listItem notActive"
              >
                Tokyo
              </li>
            </ul>
          </nav>
        </header>
        <div
          class="container flexColumn transparentBG border borderRadius shadow container"
        >
          <p
            class="default large weightNormal"
          >
            Error
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`WeatherAppTemplate renders correctly with no cities 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container full transparentBG contentWrapper"
      >
        <div
          class="container transparentBG maxWidthLarge"
        >
          <header
            class="header"
          >
            <nav>
              <ul
                class="list"
              >
                <li
                  class="listItem active"
                >
                  Ottawa
                </li>
                <li
                  class="listItem notActive"
                >
                  Moscow
                </li>
                <li
                  class="listItem notActive"
                >
                  Tokyo
                </li>
              </ul>
            </nav>
          </header>
          <div
            class="container transparentBG"
          >
            Please select a city!
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container full transparentBG contentWrapper"
    >
      <div
        class="container transparentBG maxWidthLarge"
      >
        <header
          class="header"
        >
          <nav>
            <ul
              class="list"
            >
              <li
                class="listItem active"
              >
                Ottawa
              </li>
              <li
                class="listItem notActive"
              >
                Moscow
              </li>
              <li
                class="listItem notActive"
              >
                Tokyo
              </li>
            </ul>
          </nav>
        </header>
        <div
          class="container transparentBG"
        >
          Please select a city!
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`WeatherAppTemplate renders correctly with one city 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container full transparentBG contentWrapper"
      >
        <div
          class="container transparentBG maxWidthLarge"
        >
          <header
            class="header"
          >
            <nav>
              <ul
                class="list"
              >
                <li
                  class="listItem active"
                >
                  Ottawa
                </li>
                <li
                  class="listItem notActive"
                >
                  Moscow
                </li>
                <li
                  class="listItem notActive"
                >
                  Tokyo
                </li>
              </ul>
            </nav>
          </header>
          <div
            class="container flexColumn transparentBG border borderRadius shadow container"
          >
            <p
              class="default large weightNormal"
            >
              Error
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container full transparentBG contentWrapper"
    >
      <div
        class="container transparentBG maxWidthLarge"
      >
        <header
          class="header"
        >
          <nav>
            <ul
              class="list"
            >
              <li
                class="listItem active"
              >
                Ottawa
              </li>
              <li
                class="listItem notActive"
              >
                Moscow
              </li>
              <li
                class="listItem notActive"
              >
                Tokyo
              </li>
            </ul>
          </nav>
        </header>
        <div
          class="container flexColumn transparentBG border borderRadius shadow container"
        >
          <p
            class="default large weightNormal"
          >
            Error
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
